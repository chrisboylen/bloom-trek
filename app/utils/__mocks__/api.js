export const getAllParks = () => (
  [{
      "id": 1,
      "name": "Castlewood Canyon",
      "coords": {
        "latitude": 39.3379,
        "longitude": -104.7512
      }
    },
    {
      "id": 2,
      "name": "Deer Creek Canyon",
      "coords": {
        "latitude": 39.542,
        "longitude": -105.1491
      }
    }]
);

export const getParkTrails = () => (
  [{
      "id": 10,
      "name": "Blue Loop and Challenge Hill",
      "uid": 7010867,
      "difficulty": "hard",
      "length": 0.6,
      "status": "All Clear",
      "trail_img_url": "https://cdn-files.apstatic.com/hike/7007993_medium_1446301088.jpg",
      "trail_url": "https://www.hikingproject.com/trail/7010867/blue-loop-and-challenge-hill",
      "summary": "Challenge Hill is a 200-step climb up the face of the mountain.",
      "latitude": 39.3678,
      "longitude": -104.8787,
      "park_id": 1
    }]
);

export const getFlowersByMonth = () => (
  [{
      "id": 1,
      "common_name": "blue flax",
      "scientific_name": "Adenolinum lewisii",
      "flower_img_url": "http://extension.colostate.edu/county/jeffco/natural/plant_images/adenolinum_lewisii_1_378x400.jpg",
      "description": "stigma capitate (headed); plant produces many branches from the base; styles of same length on different plants.",
      "bloom_start": 5,
      "bloom_end": 8,
      "name": "WILD BLUE FLAX",
      "habitat": "dry slopes, forest clearings, roadsides."
    },
    {
      "id": 4,
      "common_name": "purple milk vetch",
      "scientific_name": "Astragalus agrestis",
      "flower_img_url": "http://extension.colostate.edu/county/jeffco/natural/plant_images/astragalus_agrestis_400x327.jpg",
      "description": "blunt keel; flowers erect, clustered in ovoid heads; pubescence basifixed (hairs attached at the base).",
      "bloom_start": 3,
      "bloom_end": 5,
      "name": "FIELD MILKVETCH",
      "habitat": "grasslands on the plains and mesas."
    }]
);
