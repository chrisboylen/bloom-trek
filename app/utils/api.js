export const getAllParks = async () => {
  const response = await fetch(
    'https://bloom-trek-api.herokuapp.com/api/v1/parks/',
  );
  return response.json();
};

export const getParkTrails = async () => {};
