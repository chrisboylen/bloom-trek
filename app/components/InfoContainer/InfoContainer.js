import React, { Component } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { getParkTrails } from '../../utils/api';
import TrailList from '../TrailList/TrailList';
import Nav from '../Nav/Nav';

export default class InfoContainer extends Component {
  constructor() {
    super();
    this.state = {
      selectedIndex: 0,
      trails: [],
    };
  }

  updateIndex = async (selectedIndex) => {
    const { resetMap, selectedPark } = this.props;

    this.setState({ selectedIndex });
    if (selectedIndex === 1) {
      resetMap();
      this.setState({ trails: [] });
    }
    if (selectedIndex === 2 && !this.state.trails.length) {
      const trails = await getParkTrails(selectedPark);
      this.setState({ trails });
    }
  };

  render() {
    const { selectedIndex, trails } = this.state;

    return (
      <View style={styles.container}>
        {selectedIndex === 0 ? <Text style={styles.list}>{selectedIndex}</Text> : <View />}
        {selectedIndex === 2 ? <TrailList style={styles.list} trails={trails} /> : <View />}
        <Nav updateIndex={this.updateIndex} selectedIndex={selectedIndex} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  list: {
    flex: 5,
  },
});

InfoContainer.propTypes = {
  selectedPark: PropTypes.number,
  resetMap: PropTypes.func,
};
