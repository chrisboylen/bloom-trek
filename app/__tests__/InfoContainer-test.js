import React from 'react';
import { shallow } from 'enzyme';
import InfoContainer from '../components/InfoContainer';
import { getFlowersByMonth } from '../utils/api';

describe('InfoContainer', () => {
  let mockSelectedMonth;
  let mockFlowers;
  let wrapper;

  beforeEach(() => {
    mockSelectedMonth = 5;
    mockFlowers = [{
      id: 71,
      flower_img_url: 'http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg',
      name: 'Butter And Eggs',
      common_name: 'Yellow Toadflax',
      scientific_name: 'Linaria vulgaris',
      description: 'Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).',
      bloom_start: 6,
      bloom_end: 10,
      habitat: 'Open areas around former homesteads.',
    }];
    wrapper = shallow(
      <InfoContainer
        flowers={mockFlowers}
        selectedMonth={mockSelectedMonth}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('resetSelectedFlower', () => {
    it('should set state when invoked', () => {
      wrapper.setState({ selectedFlower: 1 });
      expect(wrapper.state().selectedFlower).toEqual(1);

      wrapper.instance().resetSelectedFlower();

      expect(wrapper.state().selectedFlower).toEqual(null);
    });
  });
});
