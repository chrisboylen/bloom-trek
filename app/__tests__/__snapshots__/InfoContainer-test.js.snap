// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoContainer
    flowers={
      Array [
        Object {
          "bloom_end": 10,
          "bloom_start": 6,
          "common_name": "Yellow Toadflax",
          "description": "Leaves glabrous (smooth, hairless) and glaucous (contains white waxy covering).",
          "flower_img_url": "http://extension.colostate.edu/county/jeffco/natural/plant_images/linaria_vulgaris_298x400.jpg",
          "habitat": "Open areas around former homesteads.",
          "id": 71,
          "name": "Butter And Eggs",
          "scientific_name": "Linaria vulgaris",
        },
      ]
    }
    selectedMonth={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "backgroundColor": "white",
              "borderColor": "transparent",
              "height": "85%",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": -3,
                "width": 0,
              },
              "shadowOpacity": 0.3,
            }
          }
        >
          <Component />
          <Component />
          <Component />
          <Component />
        </Component>,
        <Nav
          selectedIndex={undefined}
          updateIndex={undefined}
        />,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component />,
            <Component />,
            <Component />,
            <Component />,
          ],
          "style": Object {
            "backgroundColor": "white",
            "borderColor": "transparent",
            "height": "85%",
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": -3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "selectedIndex": undefined,
          "updateIndex": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "transparent",
                "height": "85%",
                "shadowColor": "black",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
              }
            }
          >
            <Component />
            <Component />
            <Component />
            <Component />
          </Component>,
          <Nav
            selectedIndex={undefined}
            updateIndex={undefined}
          />,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component />,
              <Component />,
              <Component />,
              <Component />,
            ],
            "style": Object {
              "backgroundColor": "white",
              "borderColor": "transparent",
              "height": "85%",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": -3,
                "width": 0,
              },
              "shadowOpacity": 0.3,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "selectedIndex": undefined,
            "updateIndex": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
